src/python/pants/option/options_bootstrapper.py

```diff
@@ -186,6 +184,11 @@ def filecontent_for(path: str) -> FileContent:
             # remote pants runners.
             os.environ["PANTS_BIN_NAME"] = bootstrap_option_values.pants_bin_name
 
+            env_tuples = tuple(
```

Comment:
@stuhood I think this keeps the spirit of #7312 in tact.

Reply:
It should as long as the `Config` discards any `env` values that don't get used (i.e., the entire `env` doesn't end up in structural equality). It looks like it should, since only the default calculations conume it...?

Not sure if worth a comment there to that effect though.